articles/plagiarism-detection/index.md

```diff
@@ -0,0 +1,402 @@
+
+We have limited our quantitative analysis scope by only looking at three key-words: **'inheritance'**, **'class'** and **'object'**. We have briefly talked about cosine similarity but I want to explain what it means in graphical terms.
+
+To explain what I mean, here is a 3D projection of the 3 documents.
+
+#### 3D Document Projection
+![3D Projection](/engineering-education/plagiarism-detection/3d-projection-plagarism-blog.png)
+Image Source: Author -- Earl Potters
+
+As illustrated by this example, we can see that `doc 2` and `doc 3` are closer in orientation by magnitude. Conversely, `doc 1` and `doc 3` are the closest using the Euclidean distance metric.
+
+The graphical representation of common words is a intersection which can be seen as a Venn diagram.
+
+#### Document Intersection
+![total common words](/engineering-education/plagiarism-detection/ven-diagrahm.png)
+Image Source: Author -- Earl Potters
+
+You can see a 3 intersecting circles which contain the set of all words in their document space. The intersection of circles is the intersection of both word sets.
+
+#### Compare and Contrast
+As you can figure out, all 3 similarity metrics have their own interpretation of similarity.
+
+- Cosine similarity checks the **orientation** of two documents
+- Euclidean distance checks the **distance magnitude** of two documents
+- Total common words checks the **intersection** of two documents.
+
+Looking at the example we can see why cosine similarity is a good metric for judging similarity, rather than using total common words or Euclidean distance.
+
+The reason for that is because total common words is very biased on file sizes while Euclidean distance is biased when comparing two different document sizes.
+
+### Plagiarism Detector Code
+Now we are at the coding step! By now you should know the following things:
+
+- A document can be converted into a **Vector space model** where the documents are represented as vectors, mostly determined by term frequency.
+- **Cosine Similarity** is an operation on vectors that will allow us to determine the similarity of two documents.
+- We can display a **normalized numerical value** between 0-1 that indicates the similarity between two documents.
+
+#### Getting Started
+In this short overview I will assume you are on an **Ubuntu** Linux operating system with **Conda** package manager.
+
+##### 1. Installing Dependencies
+You we need to install the following: `Pandas` and `scikit-learn`
+
+```Bash
+$ conda install scikit-learn pandas
+```
+
+##### 2. Verification
+To ensure that the packages are installed properly open up your python interpreter and run the follow code.
+
+``` python
+# loading python modules
+import sklearn
+import pandas as pd
+
+# Verifying version
+sklearn.show_versions()
+pd.show_versions()
+```
+> You may need to install pytest and hypothesis for pandas to run command `pd.show_versions()`
```

Comment:
If you can make a repl.it with this code, we can embed it and allow the readers to run the code right in the article. This is the preferred way to demonstrate code.